lib/pharos/configuration/bastion.rb

```diff
@@ -3,12 +3,18 @@
 module Pharos
   module Configuration
     class Bastion < Pharos::Configuration::Struct
+      attr_writer :host
+
       attribute :address, Pharos::Types::Strict::String
       attribute :user, Pharos::Types::Strict::String
       attribute :ssh_key_path, Pharos::Types::Strict::String
 
+      def ==(other)
+        address == other.address && user == other.user && ssh_key_path == other.ssh_key_path
+      end
+
       def host
-        Host.new(address: address, user: user, ssh_key_path: ssh_key_path)
+        @host ||= Host.new(address: address, user: user, ssh_key_path: ssh_key_path)
```

Comment:
`||=` with `attr_writer` feels a bit broken?

Reply:
Well yes, if someone would override the `host=` method created by `attr_writer`, it would not be called.

The "alternative" that would use the writer method would be:

```
def host
  @host || self.host = Host.new
end
```